routers/repo/editor.go

```diff
@@ -636,3 +659,37 @@ func RemoveUploadFileFromServer(ctx *context.Context, form auth.RemoveUploadFile
 	log.Trace("Upload file removed: %s", form.File)
 	ctx.Status(204)
 }
+
+// GetUniquePatchBranchName Gets a unique branch name for a new patch branch
+// It will be in the form of <lowername>-patch-<num> where <num> is the first branch of this format
+// that doesn't already exist
+func GetUniquePatchBranchName(ctx *context.Context) string {
+	prefix := ctx.User.LowerName + "-patch-"
+	for i := 1; i <= 1000; i++ {
```

Comment:
What if there are more than 1000 patch branches already? I doubt that would be likely, but what if?

Reply:
Then there will be just an empty field for the branch name and the user will have to put that in themselves, just like it is now. That is why I say it returns ""

Reply:
I added that to the function description